src/BrowserInterop/Extensions/CallBackInteropWrapper.cs

```diff
@@ -86,15 +114,31 @@ private CallBackInteropWrapper()
         public static CallBackInteropWrapper Create(Func<ValueTask> callback, object serializationSpec = null,
             bool getJsObjectRef = false)
         {
-            var res = new CallBackInteropWrapper
+            var dotnetCallback = DotNetObjectReference.Create(new JsInteropActionWrapper(callback));
+            var res = new CallBackInteropWrapper(dotnetCallback)
             {
-                CallbackRef = DotNetObjectReference.Create(new JsInteropActionWrapper(callback)),
                 SerializationSpec = serializationSpec,
                 GetJsObjectRef = getJsObjectRef
             };
             return res;
         }
 
-        public object CallbackRef { get; set; }
+        public void Dispose()
+        {
+            //casting is safe, the constructor enforces a disposable.
+            ((IDisposable)this.CallbackRef).Dispose();
+        }
+
+
+        public object CallbackRef { get;  } // we must keep it as a object, otherwise its not serialized.
+    }
+    internal interface ICallbackReferenceData
+    {
+    }
+    public class CallbackReferenceData<T>: ICallbackReferenceData
```

Comment:
Like in previous comment, this could be pulled up and use in other js interop utilities where I do get ref then get data.

Reply:
Some events like mouse move events occur really often, as described above I would prefer to avoid the call back from .NET to javascript on every event, just to get the events data.

Reply:
we'll keep the conversation on the other comment